src/compiler/parser.ts

```diff
@@ -4840,10 +4867,10 @@ namespace ts {
                 factory.createConditionalExpression(
                     leftOperand,
                     questionToken,
-                    doOutsideOfContext(disallowInAndDecoratorContext, parseAssignmentExpressionOrHigher),
+                    doOutsideOfContext(disallowInAndDecoratorContext, () => parseAssignmentExpressionOrHigher(/*allowReturnTypeInArrowFunction*/ false)),
                     colonToken = parseExpectedToken(SyntaxKind.ColonToken),
                     nodeIsPresent(colonToken)
-                        ? parseAssignmentExpressionOrHigher()
+                        ? parseAssignmentExpressionOrHigher(allowReturnTypeInArrowFunction)

```

Comment:
We had a discussion over at https://github.com/microsoft/TypeScript/pull/49531#discussion_r900542195 about whether or not consistency between the true/false branches should win, or whether backwards compatibility should win for the false case.

In other words, should the following continue to parse?

```ts
a ? b : (c): Type => c
```

While we can propagate the flag here, having the two branches differ is confusing, and could make naive refactorings fail. I have a soft vote for disallowing in conditional types, but explicitly document it.

What do you think @ahejlsberg @sandersn?

Reply:
I am pro-backwards-compatibility; this was an oversight in that I had it this way in a previous change, but while working on one of the newer examples from the wild, I switched it back until I found the "real" fix as the token === ":" check in the arrow function parsing function.

Reply:
The relevant test case is [`e76fe4c` (#49531)](https://github.com/microsoft/TypeScript/pull/49531/commits/e76fe4c669b7d9a1d041c9f191d801b805ad9d23), which I fixed in [`8675dc1` (#49531)](https://github.com/microsoft/TypeScript/pull/49531/commits/8675dc142f78c784663e814dd4b57afe8669b428) before I had a test case.